src/wp-includes/media.php

```diff
@@ -2097,10 +2097,10 @@ function _wp_post_thumbnail_class_filter_remove( $attr ) {
  *     @type string $caption    The caption text.
  *     @type string $class      Additional class name(s) added to the caption container.
  * }
- * @param string $content Shortcode content.
+ * @param string $content Shortcode content. Defaults to an empty string.
  * @return string HTML content to display the caption.
  */
-function img_caption_shortcode( $attr, $content = null ) {
+function img_caption_shortcode( $attr, $content = '' ) {
```

Comment:
In doing a [search of themes/plugins](https://wpdirectory.net/search/01FFGA3R7X5984YE5E1W2T0RPN) using this function, there are instances where `null` will be passed into this function. Some examples include VaultPress (parameter default is `null`) and Get the Image (where `preg_replace()` could return `null`.

What if this change goes one step further? 

```suggestion
function img_caption_shortcode( $attr, $content = '' ) {
	if ( null === $content ) {
		$content = '';
	}
```

@jrfnl What do you think?

Reply:
I agree we could take the change further, but if we do, I think we need to accompany it by a "doing it wrong".

After all plugins/themes have their own responsibility to 1) write valid code (`null` was never a documented allowed value, the parameter was always documented as `string`) and 2) make their code compatible with PHP 8.1.

If we take the change one step further, we're hiding the deprecation warning PHP throws when the invalid `null` value is passed, which would effectively mean that plugins/themes would have no reason to make their code comply with the specifications.

Reply:
Good point. I agree.

```suggestion
function img_caption_shortcode( $attr, $content = '' ) {
	if ( null === $content ) {
		_doing_it_wrong(
			__FUNCTION__,
			__( 'Content must be a string and is not nullable.' ),
			'5.9.0'
		);	
		$content = '';
	}
```

This change though then requires changes to some of the tests.

Reply:
Point of interest here: a "doing it wrong" throws an `E_USER_NOTICE`, while the PHP 8.1 notice is `E_DEPRECATED`, Thye are often both ignored via the error reporting settings, but `E_ÜSER_NOTICE` is "higher" in the [error hierarchy](https://www.php.net/manual/en/errorfunc.constants.php) than `E_DEPRECATED`, so we would effectively be **raising** the error level with this additional change.

Reply:
Thinking about this morning.

Summary:

- This PR changes the default value from `null` to `''`
- The `E_DEPRECATED` happens on PHP 8.1, but not on < PHP 8.1
- Adding "doing it wrong" here:
   - Adds awareness as part of input validation (though in many cases will be ignored)
   - Adds a `E_USER_NOTICE` for all PHP versions
- Adding the guard protects core for when a `null` is given and it's running on PHP 8.1+

Input validation:

- This topic fits into a broader input validation conversation which is outside the scope of this PR and ticket. 
- Without a broader input validation strategy, doesn't make sense to add `E_USER_NOTICE` when not on PHP 8.1+

Let's move forward with the original PR and then revisit this when we tackle input validation.